### DISCUSSION TITLE
React Native and the REST API are not working.

### DISCUSSION THREAD

**Discussion by ScreamZ** [nextauthjs/next-auth#12014] [Help]
### Environment

```
  System:
    OS: macOS 15.0.1
    CPU: (10) arm64 Apple M1 Max
    Memory: 90.36 MB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.11.1 - ~/.volta/tools/image/node/20.11.1/bin/node
    Yarn: 1.22.19 - ~/.volta/tools/image/yarn/1.22.19/bin/yarn
    npm: 10.2.4 - ~/.volta/tools/image/node/20.11.1/bin/npm
    pnpm: 9.11.0 - ~/.volta/bin/pnpm
    bun: 1.1.27 - ~/.bun/bin/bun
    Watchman: 2024.01.22.00 - /opt/homebrew/bin/watchman
  Browsers:
    Chrome: 129.0.6668.100
    Safari: 18.0.1
```

### Reproduction URL

https://github.com/ScreamZ/next-auth-bug-expo

### Describe the issue

I want to use Expo (React Native) to authenticate user through my app, for that I'm trying to get an Auth session. Currently I've been able to achieve the first part but this is not working.

My goal is :
1. Generate a oAuth URL
2. Get this URL on my app and open a inApp brower.
3. The user validate the oAuth.
4. Redirected to the proper URL on my API (next auth)
5. Which redirect through a deeplink to my app.
6. Grab the Cookies for getting the auth.

DAMN this is sooooo complicated! I'll try few more hours or just leave to lucia auth.

It's really hard to use the REST API and it really looks bugged.
It works on first call but on subsequent the is a csrf ERROR.
![image](https://github.com/user-attachments/assets/987d7726-1959-495d-877c-c83b2e4088a1)

### How to reproduce

Use [snack.expo.dev/@screamz/next-auth](https://snack.expo.dev/@screamz/next-auth) or the linked repo in github (same code)

Install Expo Go on your phone

Launch the given app with the reproduction URL on your phone

Change the endpoint of your application with Next Auth Enabled and Discord auth (or any oAuth service), but if not discord, change the code in the Expo Snack to call other provider. You can use us in the repo.

### Expected behavior

Every calls should returns the URL for the callback, currently this only happen on first time

**Comment by jancbeck** [comment]
https://github.com/t3-oss/create-t3-turbo/blob/main/apps/nextjs/src/app/api/auth/%5B...nextauth%5D/route.ts
